ynput/ayon-shotgrid
# Shotgrid: reviving archived assets is broken

## Changelog Description
This PR supports recreation of children Tasks in AYON if deleted Asset is revived in Shotgrid.


## Additional info
I don't like this solution too much, open for comments.

Main issue is that revival of assets on SG sends these events (in order) - `attribute_change` for CHILD task, `attribute_change` for Asset, `revival_entity`.
If we are processing first `attribute_change`, it wants to update non existing Task under non existing Asset.
Right now this throws new `MissingParentError` exception, that could be handled specifically. 

It fails processing event, it creates new source event on `shotgrid.event` (there is no way how to create new event on processing topic, imho), which gets put on the end of the queue. 
Then asset gets recreated, then this new event gets reprocessed (only once), this time hopefully it finds parent asset in AYON, so it finishes successfully (or fails forever).


## Testing notes:

1. create asset and task(s) in AYON
2. let them be synchronized to SG (easiest to run in this repo `service_tools/manage.ps` services`)
3. delete asset in AYON Editor
4. let it delete asset in SG
5. go to trash (top right your profile, Trash at the bottom, Asset)
![image](https://github.com/user-attachments/assets/ef68c0a1-6ead-4ae0-adcd-225f6a110718)
6. check if asset and tasks under it got recreated in AYON
